table/block_based/block_based_table_factory.cc

```diff
@@ -429,8 +422,16 @@ BlockBasedTableFactory::BlockBasedTableFactory(
   InitializeOptions();
   RegisterOptions(&table_options_, &block_based_table_type_info);
 
-  if (table_options_.reserve_table_reader_memory &&
-      table_options_.no_block_cache == false) {
+  const auto options_overrides_iter =
+      table_options_.cache_usage_options.options_overrides.find(
+          CacheEntryRole::kBlockBasedTableReader);
+  const auto table_reader_charged =
+      options_overrides_iter !=
+              table_options_.cache_usage_options.options_overrides.end()
```

Comment:
The InitializeOptions() logic looks like it populates entries for every `CacheEntryRole` including filling in fallbacks to default options. In that case can we use `table_options_.cache_usage_options.options_overrides.at(CacheEntryRole::kBlockBasedTableReader).charged` to get the final result directly? (Same question applies to similar code like for calculating `filter_construction_charged`.)

Reply:
For `CacheEntryRole::kBlockBasedTableReader`, yeah we should remove (and already fixed it)
For `filter_construction_charged`, some tests directly initialize with FilterBitsBuilder without going through InitializeOptions like https://github.com/facebook/rocksdb/blob/7.2.fb/util/bloom_test.cc#L301 so we can't do so without fixing a lot of the tests (so I will leave it as it is)
For dictionary compression buffer, still going through CI check and see if there is any test we need to fix for that.

Reply:
ok with me. Sounds like an issue with the tests if they are using internal APIs in a way that could not happen in non-test code.